test/binaryen.js/kitchen-sink.js

```diff
@@ -539,8 +539,8 @@ function test_core() {
     // Reference types
     module.ref.is_null(module.ref.null(binaryen.externref)),
     module.ref.is_null(module.ref.null(binaryen.funcref)),
-    module.ref.is_null(module.ref.func("kitchen()sinker")),
-    module.select(temp10, module.ref.null(binaryen.funcref), module.ref.func("kitchen()sinker"), binaryen.funcref),
+    module.ref.is_null(module.ref.func("kitchen()sinker", binaryen.funcref)),
```

Comment:
We should prefer to give `ref.func` a precise type rather than just `funcref`, to the point that we might want to consider enforcing that in the API. Being imprecise with the type always leaves useful information on the table.

Reply:
Agreed, but there is no way to do that atm in the JS API?

Reply:
Yes, the JS API only has a mechanism to create Tuple types currently, via `createType` (and expanding via `expandType`). Something similar will be needed for functions, arrays and structs eventually. Perhaps wrappers like `TupleType` (backed by what's currently `createType`), `FunctionType`, `ArrayType` and `StructType` on top of the respective new C API?

Reply:
Do you mean `createType(SignatureType(..)) as opposed to `createSignatureType(..)`? I don't have an opinion myself, not sure what's more JS-ey...

Reply:
> Agreed, but there is no way to do that atm in the JS API?

Ah, maybe that would be good to fix first, then. I don't feel too strongly about it, though.

Reply:
I don't think it should block landing this - we can get very far in testing and using typed function references, and GC, without C or JS API. But I agree it is necessary, obviously.